tests/test_date.py

```diff
@@ -2,66 +2,135 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
+from types import MethodType
+from itertools import izip
+from collections import OrderedDict
 import unittest
 from datetime import datetime, timedelta
 
 from dateutil.relativedelta import relativedelta
+
 from mock import Mock, patch
+
 from nose_parameterized import parameterized, param
 
+from dateparser.languages import LanguageDataLoader
 from dateparser import date
 from tests import BaseTestCase
 
 
-class DateRangeTest(unittest.TestCase):
+class DateRangeTest(BaseTestCase):
+    def setUp(self):
+        super(DateRangeTest, self).setUp()
+        self.period_start = NotImplemented
+        self.period_end = NotImplemented
```

Comment:
We should list all custom attributes we define during the tests in `setUp` methods.